Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -222,6 +222,46 @@
     completionHandler(std::nullopt);
 }
 
+void WebExtensionContext::declarativeNetRequestGetMatchedRules(std::optional<WebExtensionTabIdentifier> tabIdentifier, std::optional<double> minTimeStamp, CompletionHandler<void(std::optional<Vector<WebExtensionMatchedRuleParameters>> matchedRules, std::optional<String>)>&& completionHandler)
+{
+    RefPtr tab = tabIdentifier ? getTab(tabIdentifier.value()) : nil;
+
+    if (tabIdentifier && !tab) {
+        completionHandler(std::nullopt, toErrorString(@"declarativeNetRequest.getMatchedRules()", @"tabId", @"The specified tab does not exist."));
+        return;
+    }
+
+    if (!hasPermission(_WKWebExtensionPermissionDeclarativeNetRequestFeedback)) {
+        ASSERT(hasPermission(_WKWebExtensionPermissionActiveTab));
+
+        if (!hasPermission(_WKWebExtensionPermissionTabs, tab.get())) {
+            completionHandler(std::nullopt, toErrorString(@"declarativeNetRequest.getMatchedRules()", nil, @"The 'activeTab' permission has not been granted by the user for the specified tab."));
```

Comment:
Should break out `@"declarativeNetRequest.getMatchedRules()"` into `static NSString * const apiName` since you use it twice here.